calico-vpp-agent/cmd/calico_vpp_dataplane.go

```diff
@@ -227,7 +240,10 @@ func main() {
 	case <-t.Dying():
 		log.Errorf("tomb Dying %s", t.Err())
 	}
-
+	routeWatcher.Stop()
+	if linkWatcher != nil {
+		linkWatcher.Stop()
+	}
```

Comment:
We could use the tomb to handle graceful teardown instead of calling these method